### PR TITLE
feat: fix issues with constructor method 

### PR DESCRIPTION
**Pull request by armano2**
- add missing resultType and typeParameters
- fix range of constructor identifier when modifiers are present
- constructor property in objects should have kind init
- remove dead code

### CODE REVIEW

File: src/convert.ts [hunk 1]
```diff
@@ -1007,112 +1008,92 @@ export default function convert(config: ConvertConfig): ESTreeNode | null {
...
-        );
+        if (result.type !== AST_NODE_TYPES.Property) {
+          method.typeParameters = convertTSTypeParametersToTypeParametersDeclaration(
+            node.typeParameters
+          );
+        } else {
+          result.typeParameters = convertTSTypeParametersToTypeParametersDeclaration(
+            node.typeParameters
+          );
+        }
```
Comment (armano2) [inline src/convert.ts]: babel parses it in this way, in objects `typeParameters` are assigned to Property, and in classes to methods

**Comment by JamesHenry** [comment]
Great, thanks!

**Comment by armano2** [comment]
did you read my comment?

**Comment by JamesHenry** [comment]
This one? https://github.com/JamesHenry/typescript-estree/pull/124#discussion_r247357883

**Comment by armano2** [comment]
yes, i'm just unsure if we should follow them on this one

**Pull request merged**